### DISCUSSION TITLE
Proper metadata format

### DISCUSSION THREAD

**Discussion by oscar-b** [GoogleCloudPlatform/prometheus-engine#668] [Q&A] [answered]
Our metrics expose both `UNIT` and `HELP` metadata, but this doesn't seem to be picked up by Google metrics. It displays an empty description and the unit as `1`.

Does the units have to follow the standardised Unit of Measure mentioned here? https://cloud.google.com/monitoring/mql/reference#units-of-measure

Any suggestions?

```
# TYPE test_acquire_time_seconds_total counter
# UNIT test_acquire_time_seconds_total seconds
# HELP test_acquire_time_seconds_total Number of total acquire time
test_acquire_time_seconds_total 52.565
```

**Comment by lyanco** [comment, marked as answer]
Re: # HELP, you're right in that we don't pick this up right now. We have to figure out how to ingest it.
We create metrics and set descriptions when a point is first seen, but there's no place to put description in the call. We also have to figure out what to do if descriptions conflict. We want to implement this but it'll take some time.

Re: # UNIT, I don't believe that's a standard field. Prometheus doesn't really have the concept of units, they're by convention only. All the Grafana dashboards you see in the wild are manually setting the unit on the chart itself, which is what you'll have to do in either Grafana or the Cloud Monitoring UI.

**Comment by oscar-b** [comment]
> Re: # UNIT, I don't believe that's a standard field

Ah right, we've implemented our metrics according to the OpenMetric standard, where it's standardised. Just kind of assumed that would be picked up.

Thanks for the clarification!
